src/main/java/de/tum/in/www1/artemis/service/connectors/bitbucket/BitbucketService.java

```diff
@@ -265,6 +270,29 @@ public VcsRepositoryUrl copyRepository(String sourceProjectKey, String sourceRep
         throw new BitbucketException("Max retries for forking reached. Could not fork repository " + sourceRepositoryName + " to " + targetRepositoryName);
     }
 
+    @Override
+    public VcsRepositoryUrl copyRepository(String sourceProjectKey, String sourceRepositoryName, String targetProjectKey, String targetRepositoryName) {
+        final var targetRepoSlug = targetProjectKey.toLowerCase() + "-" + targetRepositoryName.toLowerCase();
+        try {
+            var sourceRepoUrl = getCloneRepositoryUrl(sourceProjectKey, sourceRepositoryName.toLowerCase());
+            URL sourceRepositoryUrlAsUrl = new URL(sourceRepoUrl.toString());
+            // checkout the source repo
+            Repository sourceRepo = gitService.getOrCheckoutRepository(sourceRepositoryUrlAsUrl, true);
```

Comment:
what happens in the following scenario (which happens rather often)? 

1) An instructor works in the online code editor in the source template repository, commits&pushes changes.
2) The instructor also works offline in his IDE and commits&pushes changes
3) There is a merge conflict in the template repository on the Artemis server
4) Sometimes later, the instructor imports the exercise.

I would assume that `getOrCheckoutRepository(...)` will reuse the existing repository (which is in a conflict state), and will try to pull the latest changes, which will fail. Then the whole import would fail.

I think we should avoid this issue. I see the following easy fix: Instead of reusing the existing repository with `getOrCheckoutRepository(url, boolean)`, we should rather clone the repository from Bitbucket again into a new folder (this is e.g. also done when we export git repos) using the method `getOrCheckoutRepository(URL repoUrl, boolean pullOnGet, String targetPath)`. The targetPath could e.g. be `REPO_CLONE_PATH`

Reply:
In this case, we should also delete the newly cloned repository after it has been copied successfully.

Reply:
But won't the repository directly from Bitbucket be also in a conflict state unless the merge conflict was fixed? 
And in case the merge conflict was solved, is the repo on Artemis still broken? This would mean that (independent from importing) the source template repo on Artemis would not work, right?

Reply:
resolved in the latest commit